### DISCUSSION TITLE
How to generate purchase order report for certain period?

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#4916] [Q&A] [answered]
I've not used any reporting features before so think I am missing something.

Under settings I have enabled Enable generation of reports. How do you actually use this feature. I want to say for example generate a report of all purchase orders for the last month and get the total amount spent?

I've looked through the docs but it's still not clear. 

```
InvenTree Version | 0.12.0 dev
Commit Hash | 61481b4eb
Commit Date | 18/05/2023
```

**Comment by SchrodingersGat** [comment, marked as answer]
You can read about the reporting features here - https://docs.inventree.org/en/latest/report/report/

The type of report you are looking for is not readily available out of the box. The natively supported report types are more related to individual orders, e.g. creating an invoice or a work order.

Introspecting data and extract the data you are looking for is certainly possible! You could use the [python interface](https://docs.inventree.org/en/latest/api/python/python/) to extract data, or write a [plugin](https://docs.inventree.org/en/latest/extend/plugins/) which gives you the information you need

**Reply by StormTrooper** [reply]
Thanks. Good to know it possible. I'll fire up the python engine and see what I can do.
